### PR TITLE
remove maintainers checklist in PR template

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
# Motivation
maintainers are not really using it, and it produces a lot of noise when opening PRs.

# Context

it seems good enough for maintainers to follow internalised best practices.
it doesn't always work out and we skip details, but a checklist won't help if it's ignored.

# Priorities

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

**Comment by Ericson2314** [comment]
I use it. It has helped me not forget steps.

**Comment by fricklerhandwerk** [comment]
Triaged in Nix team meeting:

- @fricklerhandwerk: no one expect @Ericson2314 seems to use them
- @roberth: checking boxed that don't apply is weird
- @edolstra: maybe better to have the checklist for both author and maintainer?
  - @fricklerhandwerk: alternative: have proper contribution guidelines. from experience both here and in Nixpkgs these checklists are barely updated. people look at the code. descriptions should focus on why and how.
    - @edolstra: rights
    - @thufschmitt: not everyone reads the contributing guidelines. should have a link from the template.
      - also some properties are covered by labels
      - @fricklerhandwerk: Nixpkgs uses labels to express requirements
        - @thufschmitt: problem with labels is that they're not updated automatically
- decision: @fricklerhandwerk will add a proper checklist to contributing guides and link from the template

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -40,14 +40,27 @@ Check out the [security policy](https://github.com/NixOS/nix/security/policy).
...
    Nix maintainers follow a [structured process for reviews and design decisions](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol), which may or may not prioritise your work.
 
+   Following this checklist will make the process smoother for everyone:
+
+   - [ ] Fixes an [idea approved](https://github.com/NixOS/nix/labels/idea%20approved) issue
+   - [ ] Tests, as appropriate:
+     - Functional tests – [`tests/**.sh`](./tests)
+     - Unit tests – [`src/*/tests`](./src/)
+     - Integration tests – [`tests/nixos/*`](./tests/nixos)
+   - [ ] User documentation in the [manual](..doc/manual/src)
```
Comment (fricklerhandwerk) [inline CONTRIBUTING.md:58]:
```suggestion
   - [ ] User documentation in the [manual](./doc/manual/src)
```

**Comment by fricklerhandwerk** [comment]
Merging with suggestions implemented as discussed.

**Pull request merged**